codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/RestJson.kt

```diff
@@ -47,6 +49,48 @@ class RestJsonFactory : ProtocolGeneratorFactory<HttpBoundProtocolGenerator> {
     }
 }
 
+/**
+ * This [HttpBindingResolver] implementation mostly delegates to the [HttpTraitHttpBindingResolver] class, since the
+ * RestJson1 protocol can be almost entirely described by Smithy's HTTP binding traits
+ * (https://awslabs.github.io/smithy/1.0/spec/core/http-traits.html).
+ * The only protocol-specific behavior that is truly custom is the response `Content-Type` header, which defaults to
+ * `application/json` if not overridden.
+ */
+class RestJsonHttpBindingResolver(
+    model: Model,
+    contentTypes: ProtocolContentTypes,
+) : HttpBindingResolver {
+    private val innerResolver = HttpTraitHttpBindingResolver(model, contentTypes)
+
+    override fun httpTrait(operationShape: OperationShape): HttpTrait = innerResolver.httpTrait(operationShape)
```

Comment:
I wonder if we should add a `default` to `ProtocolContentTypes` and just handle this in `httpBindingResolver` by falling back to the default instead of `.orNull()`

Reply:
@rcoh I don't know. I liked the explicitness of having default response `Content-Type` be a property of `RestJson1` configured in its separate HTTP binding resolver class and not something all protocols would need to configure (even if implicitly).

Reply:
@rcoh I've realized I can achieve the same behavior while still using a separate class by using inheritance instead of composition, with less code. I've pushed another commit, let me know if you like this better.

Reply:
Is not setting a content type ever the right thing?

Reply:
@rcoh The RFC says 

> Is not setting a content type ever the right thing?

It shouldn't be set if the HTTP body is empty.

> Any HTTP/1.1 message containing an entity-body SHOULD include a Content-Type header field defining the media type of that body.

It's also technically conceivable for a protocol to specify that no `Content-Type` header should be set in an HTTP response under some circumstances, even if the HTTP body is not empty, since it's a SHOULD.